### PR TITLE
Feature/better collection

### PR DESCRIPTION
**Pull request by patrickshao**
- Fix issue of continuing to record when on another window.
  - Now polls and checks if window is active (3 seconds) and on first change of states, will store data
- Fix issue of continuing to record when computer is asleep
  - Now will store when computer is entering sleep state.
  - Won't store if computer was idle and entered sleep state.
- Fix issue of not updating the URL and Time when difference is too low
- Fix issue of not storing data correctly when given undefined URL.
  - Better handling of undefined URL and differentiating between an already existing website and a new one
- Comment code, explaining logic behind if statements
- Remove filter on chrome:// pages 
- Add helper functions to improve code clarity 
- Remove unused code

TL;DR A lot was done to improve the collection of data

### CODE REVIEW

**Comment by patrickshao** [comment]
- Reduce false cases that would call storeData(), specifically within onUpdated and stopped onCreated event listener.

**Comment by patrickshao** [comment]
An issue that still persists is that it is possible for the polling function to storeData at the same time as an event listener, thus double counting a certain period of time stored but chances of this is relatively low and the recording is significantly better than before
